client/packages/core/src/queryTypes.ts

```diff
@@ -110,6 +110,10 @@ type Remove$<T> = T extends object
   ? { [K in keyof T as Exclude<K, '$'>]: Remove$<T[K]> }
   : T;
 
+type Remove$NonRecursive<T> = T extends object
```

Comment:
Previously we removed the `$` recursively at the top-level `InstaQLResult` when we built up the result type. But I need the `$` when I'm figuring out which fields you selected. Now we remove it at each level as we process the query.